Docs: Templating notifications (#60109)
grafana/grafana: docs/sources/alerting/manage-notifications/template-notifications/create-edit-and-delete-message-templates.md

---
aliases:
keywords:
  - grafana
  - alerting
  - notifications
  - templates
  - create templates
  - edit templates
  - delete templates

> title: Create, edit and delete message templates

Author: I think the word `Notification templates` is still not 100% confirmed, asking the team in the weekly later today.

weight: 200
---

# Create, edit and delete message templates

Message templates can be found in the Contact points tab on the Alerting page. Here you can see a list of all your message templates. A message template can contain more than one template. For example, you might want to create a message template called `email` that contains all email templates.

> 

Author: I do want to have a sentence on this somewhere, do you have a preference where?

```
A message template can contain more than one template. For example, you might want to create a message template called `email` that contains all email templates.
```

## Create message templates

Step 1: Click the New template button:

{{< figure max-width="940px" src="/static/img/docs/alerting/unified/list-message-templates-9-3.png" caption="List message templates" >}}

> Step 2: Choose a name for the message template. This name is not the name of the template that you will use in notification policies, but the name of the template as it appears in the Contact points page. The templates that you will use in notification policies should be written in the content field. For example, the screenshot below shows a template called `email.subject`:

Reviewer: This is confusing-we should chat about behaviour and look at a way to rephrase.

> Step 2: Choose a name for the message template. This name is not the name of the template that you will use in notification policies, but the name of the template as it appears in the Contact points page. The templates that you will use in notification policies should be written in the content field. For example, the screenshot below shows a template called `email.subject`:

Reviewer: The procedure for create a template ends after naming it - shouldn't it continue with "add the content in the Content field". For information on how to write content, refer to write-templates...?

> Step 2: Choose a name for the message template. This name is not the name of the template that you will use in notification policies, but the name of the template as it appears in the Contact points page. The templates that you will use in notification policies should be written in the content field. For example, the screenshot below shows a template called `email.subject`:

Reviewer: I think you've left more screenshots in than we discussed;)

> Step 2: Choose a name for the message template. This name is not the name of the template that you will use in notification policies, but the name of the template as it appears in the Contact points page. The templates that you will use in notification policies should be written in the content field. For example, the screenshot below shows a template called `email.subject`:

Author: > This is confusing-we should chat about behaviour and look at a way to rephrase.

OK! Let's chat about it this week! 🙂  I think we need some sentence about this somewhere because it's a confusing feature of the Grafana.

> The procedure for create a template ends after naming it - shouldn't it continue with "add the content in the Content field". For information on how to write content, refer to write-templates...?

You can create an empty message template by saving it without content, but I don't think we should encourage it in the documentation. The issue here is that we might mislead the user into thinking that their template is finished when in fact its just an empty template that cannot be used.

{{< figure max-width="940px" src="/static/img/docs/alerting/unified/new-message-template-9-3.png" caption="New message template" >}}

> 
> When defining a template it is important to make sure the name is unique. For example, there should be no more than one template called `email.subject` in the same message template or across different message templates. You should also avoid defining templates with the same name as default templates such as `__subject`, `__text_values_list`, `__text_alert_list`, `default.title` and `default.message`.

Reviewer: ```suggestion

Choose unique name. For example, you shouldn't have more than one template called `email.subject` in the same message template or across different message templates. Avoid defining templates with the same name as default templates such as `__subject`, `__text_values_list`, `__text_alert_list`, `default.title` and `default.message`.
```

Author: Should I add an `a` between `Choose` and `unique`?

Author: Should I add `an` between `Choose` and `unique`?

Reviewer: yes:)

Step 3: To save the message template click the Save button.

## Edit message templates

Step 1: Find the message template that you want to edit in the list of message templates and click the Edit icon.

{{< figure max-width="940px" src="/static/img/docs/alerting/unified/list-message-templates-with-template-9-3.png" caption="List message templates" >}}

Step 2: Edit the message template just as if you were creating a new message template.

Step 3: To save the message template click the Save button.

## Delete message templates

Step 1: Find the message template that you want to delete in the list of message templates and click the Delete (Trash) icon:

{{< figure max-width="940px" src="/static/img/docs/alerting/unified/list-message-templates-with-template-9-3.png" caption="List message templates" >}}

Step 2: You will be ask to confirm that you want to delete the template. If you do want to delete the template click Yes, otherwise click Cancel:

> {{< figure max-width="940px" src="/static/img/docs/alerting/unified/delete-message-template-9-3.png" caption="Delete message template" >}}

Author: Should we mention where the Edit and Delete icons are?

> {{< figure max-width="940px" src="/static/img/docs/alerting/unified/delete-message-template-9-3.png" caption="Delete message template" >}}

Reviewer: I don't think so-should be obvious?
